rest_server/Helper.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2019-2020 Siodb GmbH. All rights reserved.
+// Use of this source code is governed by a license that can be found
+// in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+func StringToByteSize(str string) (bytes uint32, err error) {
+
+	var Uint64 uint64 = 0
+
+	if regexp.MustCompile(`^[a-zA-Z]+$`).MatchString(str[len(str)-1:]) {
+		if Uint64, err = strconv.ParseUint(str[:len(str)-1], 10, 32); err != nil {
+			return 0, err
+		}
+		switch str[len(str)-1:] {
+		case "k", "K":
+			Uint64 = Uint64 * 1024
+		case "m", "M":
+			Uint64 = Uint64 * 1024 * 1024
+		case "g", "G":
+			Uint64 = Uint64 * 1024 * 1024 * 1024
+		default:
+			return 0, fmt.Errorf("Invalid unit for parameter '%v'", str)
+		}
+	} else {
+		if Uint64, err = strconv.ParseUint(str, 10, 32); err != nil {
+			return 0, fmt.Errorf("Invalid parameter '%v'", str)
+		}
+	}
+
+	return uint32(Uint64), nil
```

Comment:
You can't just truncate it to 32-bits. Need to check that `Uint64` does not exceed maximum uint32 value (2^32 - 1). If it exceeds, return error.

Reply:
@cxxman I think it's already done with function ParseInt and its argument `bitSize `
https://golang.org/pkg/strconv/#ParseInt
isn't it?

Reply:
No.
1. You use ParseUInt, which returns uint64
2. Even if parsed number is in range of uint32, there's possibility to multiply it.
Assume you parse "8G".